File: core/src/main/scala/kafka/utils/MessageLengthChecksum.scala
package kafka.utils

import java.util.zip.Checksum

class MessageLengthChecksum() extends Checksum {
  private var length = 0L

  override def update(b: Int): Unit = length += 1

  override def getValue: Long = length & 0xFFFFFFFL

  override def update(b: Array[Byte], off: Int, len: Int): Unit = length = len

  override def reset(): Unit = 0
}

File: clients/src/main/java/org/apache/kafka/common/utils/MessageLengthChecksum.java
package org.apache.kafka.common.utils;

import java.util.zip.Checksum;

public class MessageLengthChecksum implements Checksum {
    private long length = 0L;

    public void update(int b) {
        length++;
    }

    public void update(byte[] b, int off, int len) {
        length = len;
    }

    public long getValue() {
        return length;
    }

    public void reset() {
        length = 0;
    }
}